Add check labels

dwaiting opened the pull request:
Currently, only entity labels are added to the Alerta event's attributes.

With this change, both entity and check labels are added to the attributes.

If entity and check have the same label, the check label will be appended to the entity label with a forward slash separator.

Labels with names that contain periods or dollar signs are omitted because these characters are not allowed in Alerta attribute names.

dwaiting committed:
Add check labels to event attributes

Currently, only entity labels are added to the Alerta event's attributes.

With this change, both entity and check labels are added to the attributes.

If entity and check have the same label, the check label will be appended to the entity label with a forward slash separator.

Labels with names that contain periods or dollar signs are omitted because these characters are not allowed in Alerta attribute names.

dwaiting committed:
Add labels to example event

Added timestamp because the handler errors out without one.

Added labels to entity and check to test handler code changes.

satterly commented:
Closing due to inactivity. If you believe this issue has been closed prematurely please provide more information to help progress the issue. If you do not know what is expected you can ask on [Gitter chat](https://gitter.im/alerta/chat).

dwaiting commented:
The way the code is written currently, if there are any labels associated with a check, they will not make it into the Alerta event.

With this change, the Alerta "Attributes" are populated with check and/or entity labels.

satterly commented on main.go:
```diff
@@ -119,6 +120,22 @@ func sendAlert(event *corev2.Event) error {
 
+	var attributes = make(map[string]string)
+	for key, value := range event.Entity.Labels {
+		if !strings.ContainsAny(key, ".$") {
```
Just replace "." with an underscore ("_"). That's what happens for other integrations.

dwaiting replied:
Good suggestion. I've updated the code to replace the disallowed chars rather than ignore the labels.

satterly commented on main.go:
```diff
@@ -119,6 +120,22 @@ func sendAlert(event *corev2.Event) error {
+		}
+	}
+	for key, value := range event.Check.Labels {
+		if !strings.ContainsAny(key, ".$") {
```
Same here.

satterly commented:
Can you rebase with master? That should fix the tests.

dwaiting committed:
replace periods and dollar signs in attribute keys with underscores

The handler adds the entity and check labels to Alerta event attributes. Alerta does not allow periods or dollar signs in attribute names and the API will throw an error if they are present.

To prevent these errors and allow the event to be triggered, periods and dollar signs in the entity and check label names are replaced with underscores.

satterly commented:
:+1:
